src/models/sourcecodemodel.cpp

```diff
@@ -242,3 +244,25 @@ void SourceCodeModel::setSysroot(const QString& sysroot)
 {
     m_sysroot = sysroot;
 }
+
+void SourceCodeModel::find(const QString& search, Direction direction, int offset)
+{
+    auto lines = m_sourceCodeLines;
+
+    std::rotate(lines.begin(), lines.begin() + offset, lines.end());
```

Comment:
please document: why this rotate? I don't get why that's needed as it requires a re-allocation of `lines` (it detaches when you call `.begin()` on the non-const container)

generally, isn't it enough to just do a std::find_if on the subrange given by `offset`? and the direction would just decide whether we use forward or reverse iterators?